src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -2954,17 +2954,21 @@ private static object ConvertStringToInteger(
                     return result;
                 }
 
-                if (resultType == typeof(BigInteger))
-                {
-                    // Fallback for BigInteger: manual parsing using any common format.
-                    NumberStyles style = NumberStyles.AllowLeadingSign
-                        | NumberStyles.AllowDecimalPoint
-                        | NumberStyles.AllowExponent
-                        | NumberStyles.AllowHexSpecifier;
-
-                    return BigInteger.Parse(strToConvert, style, NumberFormatInfo.InvariantInfo);
+               if (resultType == typeof(BigInteger))
+               { 
+                    // Adjust NumberStyles to remove AllowHexSpecifier for standard numeric parsing
+                    NumberStyles style = NumberStyles.AllowLeadingSign 
+                    | NumberStyles.AllowDecimalPoint 
+                    | NumberStyles.AllowExponent; 
+                    BigInteger parsedValue;
+                    
+                    if (BigInteger.TryParse(strToConvert, style, NumberFormatInfo.InvariantInfo, out parsedValue))
+                    {
+                        return parsedValue;
+                    }
+                    
+                    throw new PSInvalidCastException("Failed to convert string to BigInteger.");
```

Comment:
The code is in try-catch block where we create generic error message. So we have no need re-throw and can use BigInteger.Parse() instead of BigInteger.TryParse().